cpp/src/arrow/util/thread-pool.cc

```diff
@@ -52,6 +52,11 @@ Status ThreadPool::SetCapacity(size_t threads) {
 }
 
 size_t ThreadPool::GetCapacity() {
+  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
Is there a reason to use `unique_lock` over `lock_guard` here? I hadn't ever used `unique_lock` before, e.g. https://stackoverflow.com/questions/20516773/stdunique-lockstdmutex-or-stdlock-guardstdmutex

Reply:
[`condition_variable::wait`](http://en.cppreference.com/w/cpp/thread/condition_variable/wait) requires a `unique_lock`, and I simply re-used the same idiom all around.